bundles/paikkatietoikkuna/coordinateconversion/components/select.js

```diff
@@ -0,0 +1,168 @@
+                    if( instances[1].getValue() === "KOORDINAATISTO_MAANT_2D" ) {
+                        dropdowns[3].find('option').hide();
+                        jQuery(dropdowns[i].find('.'+instances[0].getValue()+'.'+ instances[1].getValue())).show();
+        
+                    }
+                    if( instances[1].getValue() === "KOORDINAATISTO_MAANT_3D" ) {
+                        dropdowns[3].find('option').hide();
+                        dropdowns[i].find('.'+ instances[1].getValue()).show();
+                    }
+                    if( instances[1].getValue() === "KOORDINAATISTO_SUORAK_2D" ) {
+                        dropdowns[3].find('option').hide();
+                        dropdowns[i].find('.'+ instances[1].getValue()).show();
+                    }
+                    if( instances[1].getValue() === "KOORDINAATISTO_SUORAK_3D" ) {
+                        dropdowns[3].find('option').hide();
+                        dropdowns[i].find('.'+ instances[1].getValue()).show();
+                    }
+                    // Karttaprojektiojärjestelmä special cases
+                    if( instances[2].getValue() === "KKJ_KAISTA" ) {
+                        dropdowns[3].find('option').hide();
+                        dropdowns[i].find('.'+ instances[2].getValue()).show();
+                    }
+                    if( instances[2].getValue() === "TM" ) {
+                        dropdowns[3].find('option').hide();
+                        dropdowns[i].find('.'+ instances[2].getValue()).show();
+                    } 
+                    if( instances[2].getValue() === "GK" ) {
+                        dropdowns[3].find('option').hide();
+                        dropdowns[i].find('.'+ instances[2].getValue()).show();
+                    }
+                    if( instances[3].getValue() !== "COORDSYS_DEFAULT" ) {
+                        me.instance.startingSystem = true;
+                    } else {
+                        me.instance.startingSystem = true;
+                    }
+                    if( instances[4].getValue() === "KORKEUSJ_DEFAULT") {
+                        rows.each( function ( idx, row ) {
+                            var lastCell = jQuery(this).find('td:nth-last-child(2)');
+                            if( !lastCell.hasClass('heightsystem' )) {
+                                lastCell.addClass('heightsystem');
+                            }
+                        });
+                    } else {
+                        rows.each( function ( idx, row ) {
+                            var lastCell = jQuery(this).find('td:nth-last-child(2)');
+                            lastCell.attr("contenteditable", false);
+                            lastCell.removeClass('heightsystem');
+                        })
+                    }
+                        
+                    if( instances[1].getValue() === 'KOORDINAATISTO_SUORAK_2D' ) {
+                        jQuery('.map-projection').show();
```

Comment:
What is ".map-projection" and why is it searched from the whole document? Seems dangerous.